src/shogun/lib/DataType.h

```diff
@@ -40,22 +49,75 @@ enum EStructType
 
 enum EPrimitiveType
 {
-	PT_BOOL=0,
-	PT_CHAR=1,
-	PT_INT8=2,
-	PT_UINT8=3,
-	PT_INT16=4,
-	PT_UINT16=5,
-	PT_INT32=6,
-	PT_UINT32=7,
-	PT_INT64=8,
-	PT_UINT64=9,
-	PT_FLOAT32=10,
-	PT_FLOAT64=11,
-	PT_FLOATMAX=12,
-	PT_SGOBJECT=13,
-	PT_COMPLEX128=14,
-	PT_UNDEFINED=15
+	PT_BOOL=(int)TYPE::T_BOOL,
+	PT_CHAR=(int)TYPE::T_CHAR,
+	PT_INT8=(int)TYPE::T_INT8,
+	PT_UINT8=(int)TYPE::T_UINT8,
+	PT_INT16=(int)TYPE::T_INT16,
+	PT_UINT16=(int)TYPE::T_UINT16,
+	PT_INT32=(int)TYPE::T_INT32,
+	PT_UINT32=(int)TYPE::T_UINT32,
+	PT_INT64=(int)TYPE::T_INT64,
+	PT_UINT64=(int)TYPE::T_UINT64,
+	PT_FLOAT32=(int)TYPE::T_FLOAT32,
+	PT_FLOAT64=(int)TYPE::T_FLOAT64,
+	PT_FLOATMAX=(int)TYPE::T_FLOATMAX,
+	PT_SGOBJECT=(int)TYPE::T_SGOBJECT,
+	PT_COMPLEX128=(int)TYPE::T_COMPLEX128,
+	PT_UNDEFINED=(int)TYPE::T_UNDEFINED
+};
```

Comment:
Not sure about this bit. I wanted to reuse `sg_type` without having to redefine it for new enum values.